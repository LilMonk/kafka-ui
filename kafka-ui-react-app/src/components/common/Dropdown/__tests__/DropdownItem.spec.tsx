import React from 'react';
import DropdownItem from 'components/common/Dropdown/DropdownItem';
import { render } from 'lib/testHelpers';
import { fireEvent } from '@testing-library/dom';

const onClick = jest.fn();

describe('DropdownItem', () => {
  it('matches snapshot', () => {
    const wrapper = render(
      <DropdownItem onClick={jest.fn()}>Item 1</DropdownItem>
    );
    expect(onClick).not.toHaveBeenCalled();
    expect(wrapper.baseElement).toMatchSnapshot();
  });

  it('handles Click', () => {
    const wrapper = render(
      <DropdownItem onClick={onClick}>Item 1</DropdownItem>
    );

    const dropDown = wrapper.getByText('Item 1');

    fireEvent.click(dropDown);
    expect(onClick).toHaveBeenCalled();
  });
});
