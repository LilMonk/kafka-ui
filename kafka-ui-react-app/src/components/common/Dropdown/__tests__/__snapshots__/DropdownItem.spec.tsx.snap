// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownItem matches snapshot 1`] = `
<body>
  <div>
    <a
      class="dropdown-item is-link"
      href="#end"
      role="menuitem"
      type="button"
    >
      Item 1
    </a>
  </div>
</body>
`;
